src/coreclr/jit/morph.cpp

```diff
@@ -9220,11 +9220,11 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
 
             // TODO #4104: there are a lot of other places where
             // this condition is not checked before transformations.
-            if (fgGlobalMorph)
+            noway_assert(op2);
```

Comment:
not sure these noway_assert make a lot of sense but ok, the pre-existed

Reply:
Wanted to keep those as is, they're a bit better for accidental nulls anyway.